scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,186 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    nobs1, nobs2 : int
+        Sample size observed
+    diff : int of float, optional
+        The difference of mean between two samples under null hypothesis
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Which alternative hypothesis to test
+
+    Returns
+    -------
+    statistic : float
+        The test statistic calculated from observed samples
+    pvalue : float
+        The associated p-value based on estimated p-value of the standardized
+        difference.
+
+    Notes
+    -----
+    The Poisson distribution is commonly used to model many processes such as
+    transactions per user. A simple test to compare difference between two
+    means of Poisson samples is C-test, based on conditional distribution.
+    Meanwhile the E-test is an unconditional test
+
+    Based the author results [1]_, E-test is more powerful than C-test. The E-test
+    is almost exact because the test exceed the nominal value only by negligible
+    amount. Compared to C-test which produce smaller size than nominal value.
+
+    References
+    ----------
+    .. [1]  Krishnamoorthy, K., & Thomson, J. (2004). A more powerful test for
+       comparing two Poisson means. Journal of Statistical Planning and Inference,
+       119(1), 23-35.
+    .. [2]  Przyborowski, J., & Wilenski, H. (1940). Homogeneity of results in
+       testing samples from Poisson series: With an application to testing clover
+       seed for dodder. Biometrika, 31(3/4), 313-323.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Taken from Przyborowski and Wilenski (1940) [2]_. Suppose that a purchaser wishes to
+    test the number of dodder seeds (a weed) in a sack of clover seeds that he bought
+    from a seed manufacturing company. A 100 g sample is drawn from a sack of clover
+    seeds prior to being shipped to the purchaser. The sample is analyzed and found to
+    contain no dodder seeds; that is, k1 = 0. Upon arrival, the purchaser also draws
+    a 100 g sample from the sack. This time, three dodder seeds are found in the sample;
+    that is, k2 = 3. The purchaser wishes to determine if the difference between the
+    samples could not be due to chance.
+
+    >>> stats.poisson_means_test(0, 100, 3, 100)
+    PoissonMeansTestResult(statistic=-1.7320508075688772, pvalue=0.08837900929018155)
+    >>> stats.ttest_ind_from_stats(mean1=0, std1=0, nobs1=100, equal_var=False,
+    ...                            mean2=3.0/100, std2=np.sqrt(.03), nobs2=100)
+    Ttest_indResult(statistic=-1.7320508075688772, pvalue=0.0863790757063214)
+
+    The result above show evidence that the difference between two samples is significant
+    at the level of significance 0.1. Both t-test and e-test show similar p-value, but
+    will be different if the sample size is small
+
+    >>> stats.poisson_means_test(0, 10, 3, 10)
+    PoissonMeansTestResult(statistic=-1.7320508075688772, pvalue=0.08837900929018155)
```

Comment:
why do we get the same p-value here, no matter if n= 10 or 100?

Reply:
i remember that this is to be expected but forgot the reason. i'll check again the paper just to make sure

Reply:
> i remember that this is to be expected but forgot the reason. i'll check again the paper just to make sure

when I use poisson-etest from Nolan's [Github](https://github.com/nolanbconaway/poisson-etest), the related p-value for `poisson_etest(0, 10, 3, 10)` is 0.024480213245881427,  the related p-value for `poisson_etest(0, 100, 3, 100)` is 0.021578695014730778. Hope this helps.

Reply:
the parameter ordering is not the same with the one in this pull request, so it should be `poisson_etest(0, 3, 10, 10)` and poisson_etest(0, 3, 100, 100)` with the output `0.08837887527956653` and `0.08837887527956653`. by comparing with the example above, the results are similar

From the code implementation alone, this is suffice to say that the same p-value is not an error. The line responsible for this output is 
`p_x1x2 = np.multiply(prob_x1, prob_x2, out=np.zeros_like(prob_x1), where=(np.abs(t_x1x2) >= np.abs(t_k1k2)))`
`prob_x1, prob_x2` is PMF from poisson and below is the graph with `nobs=10 and nobs=100` (the two have the same shape because `nlmbd_hat` calculation adjust/cancel out the sample size increase/decrease when `diff=0` and `count1+count2` doesn't change)
![Screenshot from 2021-08-12 07-00-08](https://user-images.githubusercontent.com/19750950/129118590-5ec72211-9737-49a4-a451-4f9b9c82c286.png)
so the `p_x1x2` will have the same value **if np.abs(t_x1x2) >= np.abs(t_k1k2) for n=10 or n=100 produce the same boolean elements** and indeed that is the case.

So that concludes the reason why the p-value is same

Reply:
> the parameter ordering is not the same with the one in this pull request, so it should be `poisson_etest(0, 3, 10, 10)` and `poisson_etest(0, 3, 100, 100)` with the output `0.08837887527956653` and `0.08837887527956653`. by comparing with the example above, the results are similar
> 
> From the code implementation alone, this is suffice to say that the same p-value is not an error. The line responsible for this output is
> `p_x1x2 = np.multiply(prob_x1, prob_x2, out=np.zeros_like(prob_x1), where=(np.abs(t_x1x2) >= np.abs(t_k1k2)))`
> `prob_x1, prob_x2` is PMF from poisson and below is the graph with `nobs=10 and nobs=100` (the two have the same shape because `nlmbd_hat` calculation adjust/cancel out the sample size increase/decrease when `diff=0` and `count1+count2` doesn't change)
> ![Screenshot from 2021-08-12 07-00-08](https://user-images.githubusercontent.com/19750950/129118590-5ec72211-9737-49a4-a451-4f9b9c82c286.png)
> so the `p_x1x2` will have the same value **if np.abs(t_x1x2) >= np.abs(t_k1k2) for n=10 or n=100 produce the same boolean elements** and indeed that is the case.
> 
> So that concludes the reason why the p-value is same

Oh! I made a big mistake here! You're right!

Reply:
Oh! I made a big mistake here! You're right!

Reply:
> why do we get the same p-value here...

The `count1` is the number of events _total_ in `nobs1` observations, not the number of events per observation. The size of observation we choose (100 observations of 1g each, 10 observations of 10g each, or 1 observation of 100g) should not affect the total number of events we observe in 100g of seeds.

More information [here](https://github.com/scipy/scipy/pull/10561#discussion_r688987209).

Reply:
Ah, right! we can interpret the number of observations in any unit we want